plugins/pulp_rpm/plugins/distributors/yum/metadata/repomd.py

```diff
@@ -32,6 +36,18 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         if any((exc_type, exc_val, exc_tb)):
             os.unlink(self.metadata_file_path)
 
+    def finalize(self):
+
+        if self.gpg_sign:
+            pipe = subprocess.Popen('gpg --yes --detach-sign --armor %(metadata_file_path)s' %
+                                    {'metadata_file_path': self.metadata_file_path},
+                                    shell=True, stdout=subprocess.PIPE,
+                                    stderr=subprocess.PIPE)
+            stdout, stderr = pipe.communicate()
+            if pipe.returncode != 0:
+                raise PulpCodedException(error_codes.RPM0001, command='gpg', stdout=stdout,
```

Comment:
Would you want to create a new coded exception for an error encountered during signing of metadata? It would be a new entry here:  https://github.com/pulp/pulp_rpm/blob/master/plugins/pulp_rpm/plugins/error_codes.py#L38

I think that would be good because this is a specific place where an error could occur. Add it if you want; it's an idea not a required change.